atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -324,15 +322,18 @@ private synchronized void refreshPool() {
     }
 
     @VisibleForTesting
-    void setServersInPoolTo(Set<CassandraServer> desiredServers) {
-        Set<CassandraServer> cachedServers = getCachedServers();
-        Set<CassandraServer> serversToAdd = ImmutableSet.copyOf(Sets.difference(desiredServers, cachedServers));
-        Set<CassandraServer> absentServers = ImmutableSet.copyOf(Sets.difference(cachedServers, desiredServers));
+    void setServersInPoolTo(ImmutableSet<CassandraServer> desiredServers) {
+        ImmutableSet<CassandraServer> cachedServers = getCachedServers();
+        ImmutableSet<CassandraServer> serversToAdd =
+                ImmutableSet.copyOf(Sets.difference(desiredServers, cachedServers));
```

Comment:
it's a shame there isn't a nice method for `Sets.difference` that returns an immutable set if both inputs are also immutable

Reply:
Now that this is a lazy view of 2 immutable sets, we could get rid of the copy.